src/macros.rs

```diff
@@ -43,11 +81,13 @@ macro_rules! log {
         }
     });
 
-    // log!(target: "my_target", Level::Info, "a {} event", "log");
-    (target: $target:expr, $lvl:expr, $($arg:tt)+) => ({
+    // log!(logger: my_logger, target: "my_target", Level::Info, "a {} event", "log");
+    (logger: $logger:expr, target: $target:expr, $lvl:expr, $($arg:tt)+) => ({
         let lvl = $lvl;
         if lvl <= $crate::STATIC_MAX_LEVEL && lvl <= $crate::max_level() {
+            let logger = $logger;
```

Comment:
Also, maybe move `$logger` to the argument position? So that it can support temporary expressions such as `std::convert::identity(&MyLogger::new())`;

Reply:
Let me give it a try ..